.github/scripts/integration-test-matrix.js

```diff
@@ -0,0 +1,90 @@
+module.exports = ({ context }) => {
+  const defaultPythonVersion = "3.8";
+  const supportedPythonVersions = ["3.6", "3.7", "3.8", "3.9"];
+  const supportedAdapters = ["snowflake", "postgres", "bigquery", "redshift"];
+
+  if (context.eventName.includes("pull_request")) {
+    // if PR, generate matrix based on files changed and PR labels
+    const changes = JSON.parse(process.env.CHANGES);
+    const labels = context.payload.pull_request.labels.map(({ name }) => name);
+    console.log("labels", labels);
+    console.log("changes", labels);
+    const testAllLabel = labels.includes("test all");
+    const include = [];
+
+    for (const adapter of supportedAdapters) {
+      if (
+        changes.includes(adapter) ||
+        testAllLabel ||
+        labels.includes(`test ${adapter}`)
+      ) {
+        for (const pythonVersion of supportedPythonVersions) {
+          if (
+            pythonVersion === defaultPythonVersion ||
```

Comment:
This will only run integration tests for python3.8, unless you apply labels to PR to test other python versions (`test python3.6`). I'm not sure if this is a good default, so I'm happy to change this!

Reply:
I'm good with defaulting to just 3.8 for PRs and have all the version run for a schedule run or a commit